images/bridge2/cmd/bridge/main.go

```diff
@@ -523,22 +527,142 @@ func (m *Main) loadRequestSession(ctx context.Context, w http.ResponseWriter, r
 	return sess
 }
 
+func (m *Main) sessionCommandSource(sess *Session) commands.Source {
+	return commands.NewStaticSource([]commands.Entry{
+		{Name: "add_assistant",
+			Def: commands.Def{
+				Description: `add_assistant(name: str) -> bool
+			Add an assistant to the session.
+
+			Args:
+				name (str): The assistant's name.
+
+			Returns:
+				success (bool): True if the assistant was added successfully.`,
+				Parameters: commands.FieldDefs{
+					"name": {
+						Name:        "name",
+						Type:        "string",
+						Description: "The assistant's name",
+					},
+				},
+				Returns: nil,
+			},
+			Run: func(ctx context.Context, args commands.Fields) (commands.Fields, error) {
+				name := args.String("name")
+				name = strings.ToLower(name)
+				prompt, err := assistant.DefaultPromptTemplateForName(name)
+				if err != nil {
+					return commands.Fields{
+						"success": false,
+					}, err
+				}
+				assistant.AddAssistant(sess.SpanNow(), name, prompt)
+				return commands.Fields{
+					"success": true,
+				}, nil
+			},
+		},
+
+		{Name: "working_set_add_url",
+			Def: commands.Def{
+				Description: `working_set_add_url(url: str) -> bool`,
+				Parameters: commands.FieldDefs{
+					"url": {
+						Name:        "url",
+						Type:        "string",
+						Description: "URL to add to working set",
+					},
+				},
+				Returns: nil,
+			},
+			Run: func(ctx context.Context, args commands.Fields) (commands.Fields, error) {
+				url := args.String("url")
+				workingset.AddURL(sess.SpanNow(), url)
+				return commands.Fields{
+					"success": true,
+				}, nil
+			},
+		},
+
+		{Name: "working_set_list",
+			Def: commands.Def{
+				Description: `working_set_list_urls() -> []string`,
+				Parameters:  commands.FieldDefs{},
+				Returns:     nil,
+			},
+			Run: func(ctx context.Context, args commands.Fields) (commands.Fields, error) {
+				urls := workingset.ActiveURLs(sess.Session)
+				return commands.Fields{
+					"urls": urls,
+				}, nil
+			},
+		},
+	})
+}
+
+type unreliableHTTPSource struct {
+	commands.Source
+}
```

Comment:
This adds some sanity checks around the HTTP source for URLs that aren't REFLECT-aware.
Some of this may be useful in `pkg/commands`, but for now I only put the more general HTTP client code there to leave the error handling in bridge.